catroid/src/test/java/org/catrobat/catroid/test/content/actions/WebRequestActionTest.java

```diff
@@ -0,0 +1,193 @@
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.test.content.actions;
+
+import com.badlogic.gdx.utils.GdxNativesLoader;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.ActionFactory;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.SingleSprite;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.actions.WebRequestAction;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.test.MockUtil;
+import org.catrobat.catroid.web.WebConnection;
+import org.catrobat.catroid.web.WebConnectionFactory;
+import org.catrobat.catroid.web.WebConnectionHolder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static junit.framework.Assert.assertEquals;
+
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(GdxNativesLoader.class)
+public class WebRequestActionTest {
+
+	private static final Double ERROR_TIMEOUT = 504d;
+	private static final Double ERROR_TOO_MANY_REQUESTS = 429d;
+	private static final Double ERROR_BAD_REQUEST = 400d;
+	private static final String TEST_URL = "https://catroid-test.catrob.at/pocketcode/";
+	private static final String TEST_USERVARIABLE = "testUservariable";
+
+	private Sprite testSprite;
+	private Project project;
+	private UserVariable userVariable;
+	private WebConnectionFactory webConnectionFactory;
+	private WebConnection webConnection;
+
+	@Before
+	public void setUp() throws Exception {
+		PowerMockito.mockStatic(GdxNativesLoader.class);
+		testSprite = new SingleSprite("testSprite");
+		project = new Project(MockUtil.mockContextForProject(), "testProject");
+		ProjectManager.getInstance().setCurrentProject(project);
+		userVariable = new UserVariable(TEST_USERVARIABLE);
+		project.addUserVariable(userVariable);
+
+		StageActivity.webConnectionHolder = Mockito.mock(WebConnectionHolder.class);
```

Comment:
when mocking static fields you should really at least set them to null in teardown after the test. (or this mock stays there for the remaining test run, including all tests coming later...)